File: docs/docs/API-Reference/javascript-examples/api-projects/import-a-project.js
const url = `${process.env.LANGFLOW_URL ?? ""}/api/v1/projects/upload/`;

const formData = new FormData();
// Replace with a File/Blob for "20241230_135006_langflow_flows.zip" in your environment.
formData.append("file", new Blob(["<file contents>"]), "20241230_135006_langflow_flows.zip");

const options = {
  method: 'POST',
  headers: {
    "accept": `application/json`,
    "x-api-key": `${process.env.LANGFLOW_API_KEY ?? ""}`,
  },
  body: formData,
};

fetch(url, options)
  .then(async (response) => {
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}`);
    }
    const text = await response.text();
    console.log(text);
  })
  .catch((error) => console.error(error));

File: docs/docs/API-Reference/python-examples/api-projects/import-a-project.py
import os

import requests

url = f"{os.getenv('LANGFLOW_URL', '')}/api/v1/projects/upload/"

headers = {
    "accept": f"application/json",
    "x-api-key": f"{os.getenv('LANGFLOW_API_KEY', '')}",
}

files = {
    "file": open("20241230_135006_langflow_flows.zip", "rb"),
}

response = requests.request("POST", url, headers=headers, files=files)
response.raise_for_status()

print(response.text)

for _f in files.values():
    if hasattr(_f, 'close'):
        _f.close()
